v1/tarball/image.go

```diff
@@ -56,46 +54,20 @@ func Image(path string, tag *name.Tag) (v1.Image, error) {
 	if err := img.loadTarDescriptorAndConfig(); err != nil {
 		return nil, err
 	}
-	return &img, nil
-}
-
-func (i *image) FSLayers() ([]v1.Hash, error) {
-	panic("not implemented")
-}
-
-func (i *image) DiffIDs() ([]v1.Hash, error) {
-	panic("not implemented")
+	return partial.UncompressedToImage(&img)
 }
 
 func (i *image) ConfigName() (v1.Hash, error) {
-	panic("not implemented")
-}
-
-func (i *image) BlobSet() (map[v1.Hash]struct{}, error) {
-	panic("not implemented")
-}
-
-func (i *image) Digest() (v1.Hash, error) {
-	panic("not implemented")
-}
-
-func (i *image) MediaType() (types.MediaType, error) {
-	manifest, err := i.Manifest()
-	if err != nil {
-		return types.MediaType(""), err
+	buf := bytes.NewBuffer(nil)
+	if err := json.NewEncoder(buf).Encode(i.config); err != nil {
+		return v1.Hash{}, err
 	}
-	return manifest.MediaType, nil
+	h, _, err := v1.SHA256(v1util.NopReadCloser(buf))
+	return h, err
 }
 
-// Manifest returns the v1.Manifest for the specified image.
-// This method memoizes the result, avoiding repeated reads from the tarball.
-func (i *image) Manifest() (*v1.Manifest, error) {
-	if i.manifest == nil {
-		if err := i.loadManifestAndBlobs(); err != nil {
-			return nil, err
-		}
-	}
-	return i.manifest, nil
+func (i *image) MediaType() (types.MediaType, error) {
```

Comment:
Do you need to implement this one here, or is it covered in the partial?

Reply:
This is part of `imageCore` because it could be either `Docker` or `OCI`.  I think we're biasing towards `Docker`, but should probably be biasing towards `OCI` since we called this `v1`...

@jonjohnsonjr WDYT?